controllers/auth_controller.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable max-len */
+/* eslint-disable import/no-extraneous-dependencies */
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+const { isError } = require('joi');
+const User = require('../models/User');
+const joiSchema = require('../validators/schemas');
+const verifyToken = require('../middleware/verify_token');
+
+const signUp = async (req, res) => {
+  const {
+    email, password, name, username,
+  } = req.body;
+  const saltRounds = 10;
+
+  try {
+    await joiSchema.validateAsync({ email: req.body.email, password: req.body.password });
+    const existingUser = await User.exists({ email });
+    if (existingUser) {
+      return res.status(406).json({ success: false, message: 'User already exists' });
+    }
+    const hashPass = await bcrypt.hash(password, saltRounds);
+    const user = await User.create({
+      email, password: hashPass, name, username,
+    });
+    const token = await jwt.sign(user.id, process.env.JWT_KEY);
+    return res.status(201).json({
+      success: true,
+      token,
+      data: user,
+    });
+  } catch (error) {
+    if (isError(error)) {
+      if (error.details[0].path.includes('password')) {
+        return res.status(405).json({ success: false, message: 'Password Malformed' });
```

Comment:
this errors has to go since the middleware is validating just 500 status code will be fine